sdks/go/pkg/beam/core/runtime/exec/combine.go

```diff
@@ -213,9 +213,7 @@ func (n *Combine) addInput(ctx context.Context, accum, key, value reflect.Value,
 		// TODO(herohde) 7/5/2017: do we want to allow addInput to be optional
 		// if non-binary merge is defined?
 
-		a := reflectx.UnderlyingType(accum)
-		b := reflectx.UnderlyingType(value)
-		return reflect.ValueOf(n.mergeFn.Call2x1(a.Interface(), b.Interface())), nil
+		return n.mergeFn.Call2x1(accum, value), nil
```

Comment:
<!--new_thread; commit:92e3bbe623ddc5f61e75f5e47103456a3ad6e0af; resolved:1-->
This right here should be a fairly big win :)